proposals/NNNN-outofproc-compiler-api-architecture.md

```diff
@@ -0,0 +1,408 @@
+<!-- {% raw %} -->
+
+# Out-of-process compiler api architecture
+
+* Proposal: [NNNN](NNNN-outofproc-compiler-api-architecture.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style api will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+This document is to propose a more detailed architecture for the approved 
+proposal: [0005](0005-inproc-outofproc-compiler-api-support.md).
+
+The c api [0008](0008-c-interface-compiler-library.md) for compiling a shader
+will be implemented using this out of process architecture.
+
```

Comment:
The "Motivation" section is missing - and I was actually looking for it:

* why are we building an out of proc system?
* what problem does using process pools and things solve?

Reply:
+1  Added.